pkg/util/node/node.go

```diff
@@ -152,10 +152,7 @@ func GetZoneKey(node *v1.Node) string {
 		return ""
 	}
 
-	// We include the null character just in case region or failureDomain has a colon
-	// (We do assume there's no null characters in a region or failureDomain)
-	// As a nice side-benefit, the null character is not printed by fmt.Print or glog
-	return region + ":\x00:" + failureDomain
```

Comment:
I am not sure why a null separator was chosen, but the statement in the comment seems important and is no longer addressed in the new code. (e.g., `x:` + `::` + `y` == `x` + `::` + `:y`)

Additionally, I am not sure what happens in an upgrade scenario when some nodes are labeled with a null and some are not. I think you'd need to label nodes both ways for a number of releases? Unless this is not stored in any API?

I think it'd be much simpler to fix the code that dumps this into a prometheus metric?